### PR TITLE
add bitbucket reporter

### PR DESCRIPTION
**Pull request by sa2ajj**
Checking #2322's coverage issues.

### CODE REVIEW

**Comment by sa2ajj** [comment]
@faulteh: I added a couple of commits on top of what you provided.

The things seem to be green now.

File: master/buildbot/reporters/bitbucket.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+            self.session.headers.update({'Authorization': 'Bearer ' + token})
+
+            bitbucket_uri = '/'.join([self._base_url, owner, repo, 'commit', sha, 'statuses', 'build'])
+
+            response = yield self.session.post(bitbucket_uri, json=body)
+            if response.status_code != 201:
+                log.msg("%s: unable to upload Bitbucket status: %s" %
+                        (response.status_code, response.content))
+            else:
```
Comment (faulteh) [inline master/buildbot/reporters/bitbucket.py]: This else branch is no longer needed and was in the wrong place anyway.
I've tested this on my production buildbot just now, minus this else branch and it works in production :)

Thanks for fixing the test for me.

File: master/docs/manual/cfg-reporters.rst [hunk 2]
```diff
@@ -1034,3 +1067,6 @@ Here's a complete example:
                 result['notify'] = (build['results'] != 0)
             return result
 
+.. spelling::
```
Comment (tardyp) [inline master/docs/manual/cfg-reporters.rst]: We usually put the spelling words inside spelling_words.txt

**Comment by sa2ajj** [comment]
Oops, my fault. I'll fix it.

**Comment by sa2ajj** [comment]
I am curious what do we have here: everything seems to be passed, but GH says "Required statuses must pass before merging".  What's missing?

**Comment by tardyp** [comment]
I had to bypass the check enforcement :-/

**Comment by sa2ajj** [comment]
@faulteh, thanks for the contribution!

**Pull request merged**